src/components/v5/common/ActionSidebar/partials/forms/ManagePermissionsForm/hooks.ts

```diff
@@ -66,7 +66,10 @@ export const useManagePermissions = (
       });
       const userRole = getRole(userPermissions);
 
-      setValue('role', userRole.permissions.length ? userRole.role : undefined);
+      setValue(
+        FIELD_NAME.ROLE,
+        userRole.permissions.length ? userRole.role : undefined,
+      );
```

Comment:
I could see the intention for using an enum to achieve a strictly typed field name approach. My concern here is that we're declaring the field name in a separate object which means there's a chance that it can be out-of-sync with the validation schema. Currently, we're inferring the type of the schema to identify the field names.

So in the case of:

```js
// Before
const decisionMethod: DecisionMethod | undefined = useWatch({
  name: DECISION_METHOD_FIELD_NAME,
});
  
 // After
const decisionMethod: DecisionMethod | undefined = useWatch({
  name: FIELD_NAME.DECISION_METHOD,
});
```

We've now essentially created 3 different sources for the field names:
1. The validation schema's inferred type
2. The current constant variables we're using on `master` i.e. `DECISION_METHOD_FIELD_NAME`
3. Then the inferred type from the validation schema itself

A better approach would be to pass the type to `useWatch`:

```js
const decisionMethod: DecisionMethod | undefined = useWatch<ManagePermissionsFormValues>({
  name: 'this will be a keyof ManagePermissionsFormValues',
})
```

Then to address the loosely typed `ActionFormRow` component, we can amend that component a little bit by adding a generic type to its interface so instead of

```js
<FormCardSelect
          name={FIELD_NAME.ROLE}
          cardClassName="max-w-[calc(100vw-2.5rem)] md:max-w-sm md:px-4 md:[&_.section-title]:px-2"
          renderSelectedValue={(option, placeholder) =>
            getRoleLabel(option?.value) || placeholder
```

We can do:

```js
<FormCardSelect<ManagePermissionsFormValues>
          name="keyof ManagePermissionsFormValues"
          cardClassName="max-w-[calc(100vw-2.5rem)] md:max-w-sm md:px-4 md:[&_.section-title]:px-2"
          renderSelectedValue={(option, placeholder) =>
            getRoleLabel(option?.value) || placeholder
```

But perhaps this will increase the overall complexity and hours of the ticket 😅